### PR TITLE
test ismonomial + small things + suggestions @fingolfin

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: docs/src/CommutativeAlgebra/ideals.md [hunk 1]
```diff
@@ -386,9 +386,16 @@ eliminate(I::MPolyIdeal{T}, l::Vector{T}) where T <: MPolyElem
...
 ```@docs
 iszero(I::MPolyIdeal)
+```
+
+```@docs
 isone(I::MPolyIdeal)
 ```
 
+```@docs
+ismonomial(f:: MPolyElem)
```
Comment (thofma) [inline docs/src/CommutativeAlgebra/ideals.md]:
```suggestion
ismonomial(f::MPolyElem)
```

File: src/Rings/mpoly-ideals.jl [hunk 2]
```diff
@@ -1165,3 +1179,62 @@ function isone(I::MPolyIdeal)
...
 end
+
+################################################################################
+#
+#  ismonomial functions
+#
+################################################################################
+
+@doc Markdown.doc"""
+    ismonomial(f:: MPolyElem)
```
Comment (thofma) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
    ismonomial(f::MPolyElem)
```

File: src/Rings/mpoly-ideals.jl [hunk 3]
```diff
@@ -1165,3 +1179,62 @@ function isone(I::MPolyIdeal)
...
+false
+
+julia> ismonomial(g)
+true
+
+julia> ismonomial(ideal(R, [f, g]))
+true
+```
+"""
+function ismonomial(f:: MPolyElem)
```
Comment (thofma) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
function ismonomial(f::MPolyElem)
```

File: src/Rings/mpoly-ideals.jl [hunk 4]
```diff
@@ -1165,3 +1179,62 @@ function isone(I::MPolyIdeal)
...
+function _ismonomial(V::Vector{<: MPolyElem})
+  return all(ismonomial, V)
+end
+
+function ismonomial(I::MPolyIdeal)
+  if (_ismonomial(gens(I)))
+    return true
+  end
+  GB = groebner_basis(I, complete_reduction = true)
+  if (_ismonomial(GB))
```
Comment (thofma) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
  if _ismonomial(GB)
```

File: src/Rings/mpoly-ideals.jl [hunk 5]
```diff
@@ -1165,3 +1179,62 @@ function isone(I::MPolyIdeal)
...
+      return false
+   end  
+end
+
+function _ismonomial(V::Vector{<: MPolyElem})
+  return all(ismonomial, V)
+end
+
+function ismonomial(I::MPolyIdeal)
+  if (_ismonomial(gens(I)))
```
Comment (thofma) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
  if _ismonomial(gens(I))
```

**Comment by thofma** [comment]
Will merge once green again.

**Comment by wdecker** [comment]
@thofma   thx

**Pull request merged**